packages/graph/hash_graph/lib/graph/src/shared/identifier/knowledge.rs

```diff
@@ -160,6 +164,15 @@ pub struct EntityEditionId {
     version: EntityVersion,
 }
 
+impl SubgraphIndex<Entity> for EntityEditionId {
+    fn subgraph_entry<'r>(
```

Comment:
What's the intended meaning behind `'r` by the way?

Reply:
I have used `'r` as it belongs to the `Record`

Reply:
Isn't `'r` constrained by the lifetime of the reference to the `s`ubgraph though?

Reply:
Yes, it's bound by the `Subgraph`, but the returned entry is a `Record`. I don't feel strongly either way, to avoid confusion we could just go with `'a`.

Reply:
I think let's go with `'a` here and similar places then.

Reply:
Changed the lifetime parameter to `'a` in 15d717be9033590d7c7fe5936f0cf19075b4887d